### DISCUSSION TITLE
transaction error: `target: basehub.-.primary: vttablet: rpc error: code = Aborted desc = transaction 1690153922245058554: in use: for query (CallerID: 02u7baiholtwux1igovt)`

### DISCUSSION THREAD

**Discussion by julianbenegas** [planetscale/discussion#502] [Q&A]
hey -- i'm getting this error, which is pretty hard to debug...

currently migrating from Postgres to PlanetScale MySQL, but getting this error and i'm a bit stuck.

happy to provide more details, but i'm not sure which query inside the transaction is failing. i'm certain i'm not getting into the 20 second limit for transactions, as this error pops up very early on in the request.

**Comment by orware** [comment]
Hi @julianbenegas,

I'll continue working with you via your ticket since you've filed one already on this topic as well.

**Comment by omarnazih** [comment]
Got same issue , any updates?

**Comment by shunkakinoki** [comment]
Have the same issue, seeking for help on this!

**Comment by pencilcheck** [comment]
Not sure if it is related but getting a lot of those errors as well
`[target: tablename.-.primary: vttablet: rpc error: code = Canceled desc = EOF (errno 2013) (sqlstate HY000)]`

**Comment by GiovanniColonni** [comment]
I'm experiencing the problem as smell while doing a lot of concurrent write to the db using MySql.
`sqlState: 'HY000',
  sqlMessage: 'target: tableName.-.primary: vttablet: rpc error: code = Aborted desc = transaction 1695217028561018330: in use: in use: for tx killer rollback (CallerID: ja0v0cwqhxv05ceauke9)'`
  
 can't be ralted to this other ticket? https://github.com/planetscale/discussion/discussions/235 or some other rate limit

**Comment by anszom** [comment]
I have noticed this error as well. In my case, it was caused by multiple requests submitted simultaneously via HTTP within one transaction. Something like this:
```
connection.transaction((tx) => {
    /* MISSING AWAIT */ tx.execute(...);
    /* MISSING AWAIT */ tx.execute(...);
}
```

In fact, the above example triggers this problem twice, because the final "COMMIT" may also run in parallel with the two queries.
